torch/csrc/jit/script/compiler.cpp

```diff
@@ -2455,7 +2456,12 @@ struct to_ir {
         return graph->insertConstant(false, nullptr, tree->range());
       } break;
       case TK_NONE: {
-        return graph->insertConstant(IValue(), nullptr, tree->range());
+        auto constant = graph->insertConstant(IValue(), nullptr, tree->range());
+        if (type_hint) {
+          AT_ASSERT(type_hint->kind() == TypeKind::OptionalType);
+          constant->setType(type_hint);
```

Comment:
Would it be better to add a `type_hint` parameter to `insertConstant`? idk if it's used elsewhere but potentially it could give future code authors hints on doing the right thing